### PR TITLE
Moved the metadata into setup.cfg.

### PR DESCRIPTION
**Pull request by KOLANICH**
+ ignored some stuff

### CODE REVIEW

**Comment by alexmojaki** [comment]
Hi, can you explain what the goal/purpose of these PRs is?

What would be the procedure to release a new version of a package?

Can you fix the travis build?

**Comment by KOLANICH** [comment]
>Hi, can you explain what the goal/purpose of these PRs is?

1. store the metadata as data, not as code. This will allow to phase out metadata stored as code in future. It will also make it more available for tools.
2. get the version from the git tag automatically.

>What would be the procedure to release a new version of a package?

1. make a new git tag `v<version>`
2. `python3 -m pep517.build .`
3. `twine upload ./dist/*.whl`

**Comment by alexmojaki** [comment]
Does this have something to do with IPython?

**Comment by KOLANICH** [comment]
Yes, I have installed a new version of ipython from git and noticed a new dependency; I have looked into it and its dependencies and disliked the way they are packaged, so I have sent PRs to them. Fortunately with the help ov setuptools_py2cfg converting the metadata was pretty easy.

File: .travis.yml [hunk 1]
```diff
@@ -13,12 +13,17 @@ env:
...
 
+before_install:
+  - pip install --upgrade coveralls setuptools>=44 setuptools_scm>=3.4.3 pep517
+
 install:
-  - pip install coveralls
-  - pip install .[tests]
+  - python3 -m pep517.build .
+  - ls -l
+  - mv ./dist/*.whl ./dist/stack_data-0.CI-py3-none-any.whl
```
Comment (alexmojaki) [inline .travis.yml]: Why this complex filename?

File: .travis.yml [hunk 2]
```diff
@@ -13,12 +13,17 @@ env:
...
+  - pip install "./dist/stack_data-0.CI-py3-none-any.whl[tests]"
 
 script:
-  - coverage run --include='stack_data/*' -m pytest
+  - coverage run --branch --include='stack_data/*' -m pytest --junitxml=./rspec.xml
```
Comment (alexmojaki) [inline .travis.yml]: Why these changes?

**Comment by alexmojaki** [comment]
Please add pep517 as a test dependency to make packaging easier, and make a shell script out of these steps you just gave.

File: .travis.yml (hunk 1 above)
Comment (KOLANICH) [reply .travis.yml]: We cannot install a wheel by mask. So we rename it into some name we can use. Wheel name must follow certain format. So I have taken the name of a generated wheel an replaced version part in it to `0.CI`

File: .travis.yml (hunk 2 above)
Comment (KOLANICH) [reply .travis.yml]: Just improvements.
1. branch coverage is more sensible metric than line coverage, so why not to compute it too?
2. `--junitxml=./rspec.xml` is just a reminder that it is possible to generate a report. Appveyor and GitLab CI can use them to draw a list of passed and failed tests. IDK about GH Actions. So when you migrate to something supporting reading the list, it makes sense to register this artifact to be used.

File: .travis.yml (hunk 1 above)
Comment (alexmojaki) [reply .travis.yml]: Wow, I'm really surprised to learn that pip refuses to install wheels with invalid names. That's very unconventional.

Can you put the filename in an environment variable so that it's not duplicated?

**Comment by KOLANICH** [comment]
>make a shell script out of these steps you just gave.

It is not trivial to read the previous tag describing a version, and increment it in bash. And versions, if they are SemVers, must be incremented by a human.

**Comment by alexmojaki** [comment]
The version can be an argument which I type in manually. I just don't want to have to remember and manually enter three commands.

**Comment by alexmojaki** [comment]
Anyway, thanks for your help and for teaching me new things. Before I merge anything, can you also make the same changes in `executing` and `pure_eval`?

File: make_release.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env sh
```
Comment (alexmojaki) [inline make_release.sh]:
```suggestion
#!/usr/bin/env bash

set -eux
```

File: make_release.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+git tag -af ${1}
```
Comment (alexmojaki) [inline make_release.sh]: Why `-a`? Remove the `-f`, this script should be safe.

File: make_release.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+git push --force  origin master ${1}
```
Comment (alexmojaki) [inline make_release.sh]: Remove the `--force`.

File: make_release.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+set -eux;
+
+if [${1+x}]; then
```
Comment (alexmojaki) [inline make_release.sh]: What does `${1+x}` mean?

**Comment by alexmojaki** [comment]
Next time I want to release a change, I will test this branch out in practice. I'm not keen to merge into master a new set of tools that I'm completely unfamiliar with.

File: make_release.sh (hunk 6 above)
Comment (KOLANICH) [reply make_release.sh]: existence of variqble

**Comment by alexmojaki** [comment]
OK, I tried using this in practice and ran into a couple of problems.

Firstly, no matter how I run the script, the argument detection fails and it tells me to provide a version:

```
$ ./make_release.sh 0.0.7
+ '[x]'
./make_release.sh: line 4: [x]: command not found
+ echo './make_release.sh <major>.<minor>.<patch>'
./make_release.sh <major>.<minor>.<patch>
+ exit 1
```

Secondly, how am I meant to work with `version.py` in practice? Someone should be able to clone this repo and then run tests, but they hit an import error because `version.py` doesn't exist. They shouldn't have to build the project to work with it.

**Comment by alexmojaki** [comment]
OK, I fixed the problems and merged in #7.

**Comment by alexmojaki** [comment]
Thank you for this contribution @KOLANICH, I've learned some new stuff!

Can you similarly update your PRs for executing and pure_eval?
